spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/ConversionUtils.java

```diff
@@ -128,7 +129,19 @@ public ByteArray convert(byte[] bytes) {
 			.of(SPANNER_TO_JAVA_DATE_CONVERTER, TIMESTAMP_INSTANT_CONVERTER, SPANNER_TO_JAVA_BYTE_ARRAY_CONVERTER,
 					SPANNER_TO_JAVA_TIMESTAMP_CONVERTER);
 
-	static Class boxIfNeeded(Class propertyType) {
+	/**
+	 * Converters to and from Spanner types.
+	 */
+	public static final Collection<Converter> DEFAULT_SPANNER_CONVERTERS = ImmutableSet
+			.<Converter>builder().addAll(DEFAULT_SPANNER_WRITE_CONVERTERS)
+			.addAll(DEFAULT_SPANNER_READ_CONVERTERS).build();
+
+	public static Set<Class> SPANNER_KEY_COMPATIBLE_TYPES = ImmutableSet.<Class>builder()
```

Comment:
I'm wondering why we needed to introduce this? Aren't these already in the mapping keySets anyway?

Reply:
Oh I see - I guess we can't have arrays for keys, that's the reason, right? It's interesting that ByteArray is supported.

Reply:
can we not have this as `public`? And maybe actually move it to the `MappingSpannerConverter` not a utility class? Same applies to `DEFAULT_SPANNER_CONVERTERS`

Reply:
changes made